services/identity-service/src/vc/vc.service.ts

```diff
@@ -0,0 +1,66 @@
+import { Injectable, InternalServerErrorException, Logger, NotFoundException } from '@nestjs/common';
+import { PrismaService } from '../utils/prisma.service';
+import * as ION from '@decentralized-identity/ion-tools';
+import { DidService } from '../did/did.service';
+import { DIDDocument } from 'did-resolver';
+import { VaultService } from '../utils/vault.service';
+import { Identity } from '@prisma/client';
+@Injectable()
+export default class VcService {
+  constructor(
+    private readonly primsa: PrismaService,
+    private readonly didService: DidService,
+    private readonly vault: VaultService,
+  ) {}
+
+  async sign(signerDID: string, toSign: string) {
+    let did: Identity;
+    try {
+      did = await this.primsa.identity.findUnique({
+        where: { id: signerDID },
+      });
+    } catch (err) {
+      Logger.error('Error fetching signerDID:', err);
+      throw new InternalServerErrorException(`Error fetching signerDID`);
+    }
+
+    if (!did) throw new NotFoundException('Signer DID not found!');
+
+    try {
+      const signedJWSEd25519 = await ION.signJws({
```

Comment:
What about RSA?

Reply:
variable name has been updated

Reply:
the algorithm is picked up automatically via the keys